### PR TITLE
[stable/elasticsearch] do not set default resources limits for all nodes types

### PR DESCRIPTION
**Pull request by fabriziofortino**
In a number of scenarios, it's better to not provide resources limits (especially regarding the CPU). Some of these scenarios are here https://github.com/kubernetes/kubernetes/issues/67577 and https://github.com/kubernetes/kubernetes/issues/51135. 

Currently, this chart defines limits for CPU that could be tricky to unset when used through CD tools (eg: argocd). Moreover, these defaults are not documented (`{}` is specified in the related defaults column).

This PR removes the defaults for resources `limits` keeping only the `requests`. The same strategy is used by various charts (eg: postgres).

- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by fabriziofortino** [comment]
@rendhalver @icereval any comment on this? 🙇

**Comment by svenmueller** [comment]
we are also waiting to get this change merged 👍

File: stable/elasticsearch/README.md [hunk 1]
```diff
@@ -63,7 +63,7 @@ The following table lists the configurable parameters of the elasticsearch chart
 | ------------------------------------ | ------------------------------------------------------------------- | --------------------------------------------------- |
 | `appVersion`                         | Application Version (Elasticsearch)                                 | `6.6.1`                                             |
 | `image.repository`                   | Container image name                                                | `docker.elastic.co/elasticsearch/elasticsearch-oss` |
-| `image.tag`                          | Container image tag                                                 | `6.5.1`                                             |
+| `image.tag`                          | Container image tag                                                 | `6.5.4`                                             |
```
Comment (rendhalver) [inline stable/elasticsearch/README.md]: This should be in a separate change.

Comment (fabriziofortino) [reply stable/elasticsearch/README.md]: @rendhalver the PR was out-of-date. I've merged it with master. It should be OK now.

**Comment by tetianakravchenko** [comment]
@hectorj2f @Towmeykaw maybe you could have a look on this PR?

**Review by hectorj2f — COMMENTED** [review]
I'd vote to remove the limits but I am not aware why these values were defined in first place. This change will also affect other people runnings the chart without cpu limits for these components.

**Comment by anas-aso** [comment]
@hectorj2f the only place where I see it relevant/dangerous is here https://github.com/helm/charts/blob/master/stable/elasticsearch/templates/data-statefulset.yaml#L126. How about introducing a new value for number of processors instead of relying on the CPU limit directly ?

**Pull request closed without merging**